src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -821,11 +821,31 @@ private enum PendingOperations
                     break;
 
                 pendingOps.Add(op);
-                meter.IncrementCounter(1);
 
+                var llt = context.Transaction.InnerTransaction.LowLevelTransaction;
+
+                var now = DateTime.UtcNow;
+                if (now - _lastHighDirtyMemCheck > TimeSpan.FromSeconds(10)) // we do not need to test scratch dirty mem every write
```

Comment:
Let's make the time configurable as well

Reply:
And default to 30 seconds, I think.
This isn't something that we are likely to need high precision for